src/date-picker/calendar.jsx

```diff
@@ -325,13 +327,11 @@ const Calendar = React.createClass({
               elementType="ul"
               style={styles.weekTitle}
             >
-              <li style={weekTitleDayStyle}>S</li>
-              <li style={weekTitleDayStyle}>M</li>
-              <li style={weekTitleDayStyle}>T</li>
-              <li style={weekTitleDayStyle}>W</li>
-              <li style={weekTitleDayStyle}>T</li>
-              <li style={weekTitleDayStyle}>F</li>
-              <li style={weekTitleDayStyle}>S</li>
+              {[...Array(7)].map((e, i) => (
```

Comment:
Could we move `[...Array(7)` out of the render method?
This will save memory allocation between two render calls.